### ISSUE TITLE
prop-types false positive outside of component and false negative in component

### ISSUE THREAD

**Issue by billyjanitsch** [yannickcr/eslint-plugin-react#485]
Ran into the following issue -- apologies if it's a duplicate; my search of existing issues didn't turn up anything similar.

``` js
export const TestComponent = props =>
  <div onClick={() => props.test()} /> // no error

export const mapStateToProps = (_, props) => ({
  otherProp: props.otherProp, // 'otherProp' is missing in props validation
})
```

Expected: `props.test()` should throw a `prop-types` error since `TestComponent` has no propTypes defined.

Also expected: the `props.otherProp` in `mapStateToProps` should *not* throw since `mapStateToProps` is not a component.

eslint: 2.2.0
eslint-plugin-react: 4.1.0
